exercises/secret-handshake/secret_handshake_test.py

```diff
@@ -1,47 +1,71 @@
 import unittest
 
-from secret_handshake import handshake, code
+from example import handshake, code
 
 
+# Tests adapted from `problem-specifications//canonical-data.json` @ v1.1.0
+
 class HandshakeTest(unittest.TestCase):
-    def test_shake_int(self):
-        self.assertEqual(handshake(9), ['wink', 'jump'])
+    def test_wink_for_1(self):
+        self.assertEqual(handshake(1), ['wink'])
+
+    def test_double_blink_for_10(self):
+        self.assertEqual(handshake(2), ['double blink'])
+
+    def test_close_your_eyes_for_100(self):
+        self.assertEqual(handshake(4), ['close your eyes'])
+
+    def test_jump_for_1000(self):
+        self.assertEqual(handshake(8), ['jump'])
+
+    def test_combine_two_actions(self):
+        self.assertEqual(handshake(3), ['wink', 'double blink'])
+
+    def test_reverse_two_actions(self):
+        self.assertEqual(handshake(19), ['double blink', 'wink'])
+
+    def test_reversing_one_action_gives_the_same_action(self):
+        self.assertEqual(handshake(24), ['jump'])
 
-    def test_shake_bin1(self):
-        self.assertEqual(
-            handshake('10110'), ['close your eyes', 'double blink'])
+    def test_reversing_no_actions_still_gives_no_actions(self):
+        self.assertEqual(handshake(16), [])
 
-    def test_shake_bin2(self):
-        self.assertEqual(handshake('101'), ['wink', 'close your eyes'])
+    def test_all_possible_actions(self):
+        self.assertEqual(handshake(15), ['wink',
+                                         'double blink',
+                                         'close your eyes',
+                                         'jump'])
 
-    def test_shake_negative_int(self):
-        self.assertEqual(handshake(-9), [])
+    def test_reverse_all_possible_actions(self):
+        self.assertEqual(handshake(31), ['jump',
+                                         'close your eyes',
+                                         'double blink',
+                                         'wink'])
 
-    def test_shake_bin_invalid(self):
-        self.assertEqual(handshake('121'), [])
+    def test_do_nothing_for_zero(self):
+        self.assertEqual(handshake(0), [])
 
-    def test_unknown_action(self):
-        self.assertEqual(code(['wink', 'sneeze']), '0')
+    # Track-specific tests
 
     def test_code1(self):
```

Comment:
Maybe we should skip tack-specific tests and let the user decide if he/she wants to write the code which passes them?

Reply:
I like that idea, since in this case the track-specific tests involve a method that isn't tested for in the canonical tests.